src/plugins/ImportKeras/ImportKeras.js

```diff
@@ -0,0 +1,307 @@
+/*globals define*/
+/*eslint-env node, browser*/
+
+/**
+ * Generated by PluginGenerator 2.20.5 from webgme on Tue Sep 10 2019 15:28:36 GMT-0500 (Central Daylight Time).
+ * A plugin that inherits from the PluginBase. To see source code documentation about available
+ * properties and methods visit %host%/docs/source/PluginBase.html.
+ */
+
+define([
+    'plugin/PluginConfig',
+    'text!./metadata.json',
+    'plugin/PluginBase',
+    'deepforge-keras/JSONModelMaps',
+    'deepforge-keras/json-model-parser'
+], function (
+    PluginConfig,
+    pluginMetadata,
+    PluginBase,
+    ModelMaps,
+    JSONLayerParser
+) {
+    'use strict';
+
+    pluginMetadata = JSON.parse(pluginMetadata);
+
+
+    /**
+     * Initializes a new instance of ImportKeras.
+     * @class
+     * @augments {PluginBase}
+     * @classdesc This class represents the plugin ImportKeras.
+     * @constructor
+     */
+    var ImportKeras = function () {
+        // Call base class' constructor.
+        PluginBase.call(this);
+        this.pluginMetadata = pluginMetadata;
+    };
+
+    /**
+     * Metadata associated with the plugin. Contains id, name, version, description, icon, configStructure etc.
+     * This is also available at the instance at this.pluginMetadata.
+     * @type {object}
+     */
+    ImportKeras.metadata = pluginMetadata;
+
+    // Prototypical inheritance from PluginBase.
+    ImportKeras.prototype = Object.create(PluginBase.prototype);
+    ImportKeras.prototype.constructor = ImportKeras;
+
+    /**
+     * Main function for the plugin to execute. This will perform the execution.
+     * Notes:
+     * - Always log with the provided logger.[error,warning,info,debug].
+     * - Do NOT put any user interaction logic UI, etc. inside this method.
+     * - callback always has to be called even if error happened.
+     *
+     * @param {function(Error|null, plugin.PluginResult)} callback - the result callback
+     */
+    ImportKeras.prototype.main = function (callback) {
+        let srcJsonHash = this.getCurrentConfig().srcModel;
+        if (!srcJsonHash) {
+            return callback(new Error('Keras Json Not Provided'), this.result);
+        }
+
+        this.archName = this.getCurrentConfig().archName;
+        return this.blobClient.getMetadata(srcJsonHash)

```

Comment:
`async`/`await` is a bit cleaner. I have been trying to move from promises to async/await when possible

Reply:
Yeah. Agreed. It was part of the generated code. This code block takes care of it.
```javascript
    ImportKeras.prototype.main = async function (callback) {
        let srcJsonHash = this.getCurrentConfig().srcModel;
        if (!srcJsonHash) {
            return callback(new Error('Keras Json Not Provided'), this.result);
        }
        try {
            this.archName = this.getCurrentConfig().archName;
            let metadata = await this.blobClient.getMetadata(srcJsonHash);
            this.archName = this.archName ? this.archName : metadata.name.replace('.json', '');
            let modelJson = await this.blobClient.getObjectAsJSON(srcJsonHash);
            this.modelInfo = JSONLayerParser.flatten(modelJson).config;
            this.addNewArchitecture();
            this.addLayers();
            await this.addConnections();
            await this.save('Completed Import Model',);
            this.result.setSuccess(true);
            return callback(null, this.result);
        }
        catch (err) {
            return callback(err, this.result);
        }
    };
```